### PR TITLE
config: add replicate-mode configuration

### PR DESCRIPTION
**Pull request by disksing**
### What problem does this PR solve? 
Fix: https://github.com/pingcap/pd/issues/2269

### What is changed and how it works?
Add config structure and adjust function.

### Check List 

Tests 
 - Unit test

### CODE REVIEW

File: server/config/config.go [hunk 1]
```diff
@@ -1115,3 +1120,36 @@ func (c DashboardConfig) ToTiDBTLSConfig() (*tls.Config, error) {
...
+	}
+	c.DRAutoSync.adjust(meta.Child("dr-autosync"))
+}
+
+// DRAutoSyncReplicateConfig is the configuration for auto sync mode between 2 data centers.
+type DRAutoSyncReplicateConfig struct {
+	LabelKey         string            `toml:"label-key" json:"label-key"`
+	Primary          string            `toml:"primary" json:"primary"`
+	DR               string            `toml:"dr" json:"dr"`
+	MasterReplicas   int               `toml:"primary-replicas" json:"primary-replicas"`
```
Comment (nolouch) [inline server/config/config.go]:
```suggestion
	PrimaryReplicas   int               `toml:"primary-replicas" json:"primary-replicas"`
```

File: server/config/config.go [hunk 2]
```diff
@@ -1115,3 +1120,36 @@ func (c DashboardConfig) ToTiDBTLSConfig() (*tls.Config, error) {
...
+	DRReplicas       int               `toml:"dr-replicas" json:"dr-replicas"`
+	WaitStoreTimeout typeutil.Duration `toml:"wait-store-timeout" json:"wait-store-timeout"`
+	WaitSyncTimeout  typeutil.Duration `toml:"wait-sync-timeout" json:"wait-sync-timeout"`
+}
+
+func (c *DRAutoSyncReplicateConfig) adjust(meta *configMetaData) {
+	if !meta.IsDefined("wait-store-timeout") {
+		c.WaitStoreTimeout = typeutil.Duration{Duration: time.Minute}
```
Comment (HunDunDM) [inline server/config/config.go]: Need to define constants like other functions?

File: server/config/config.go (hunk 1 above)
Comment (disksing) [reply server/config/config.go]: ok

File: server/config/config.go (hunk 2 above)
Comment (disksing) [reply server/config/config.go]: ok

**Pull request merged**